packages/schema-registry-api/src/api.ts

```diff
@@ -32,15 +32,17 @@ const apiFetch = async <T>(req: string | Request, init: RequestInit = {}): Promi
       'Content-Type': 'application/vnd.schemaregistry.v1+json',
     },
   };
-  const res = await fetch(req, { ...defaultInit, ...init });
+
+  let res;
   let data;
   try {
+    res = await fetch(req, { ...defaultInit, ...init });
     data = await res.json();
   } catch (error) {
-    // Generic error to avoid including URL in message, as it may contain
-    // sensitive credentials.
-    throw TypeError('Schema registry responded with invalid JSON.');
+    // Use regex to sanitize error logging in order avoid basic auth credentials leakage
+    throw TypeError(error.toString().replace(/[^/]*@/g, '{CREDENTIALS}'));
```

Comment:
Does it make sense to just return an `Error` here rather than a `TypeError`?

Reply:
Yeah makes sense, I've changed it 😉